ci/scripts/python_wheel_macos_build.sh

```diff
@@ -96,25 +96,27 @@ cmake \
     -DARROW_BUILD_SHARED=ON \
     -DARROW_BUILD_STATIC=OFF \
     -DARROW_BUILD_TESTS=OFF \
+    -DARROW_COMPUTE=ON \
+    -DARROW_CSV=ON \
     -DARROW_DATASET=${ARROW_DATASET} \
```

Comment:
Should we just hardcode this to ON? (the ARROW_PYTHON=ON would have ensure that) I don't think we want to create wheels without dataset enabled?

Reply:
> I don't think we want to create wheels without dataset enabled

I think so but `${ARROW_DATASET}` is also used for `export PYARROW_WITH_DATASET=${ARROW_DATASET}` below. So I think that we use `${ARROW_DATASET}` here for consistency. (`ARROW_DATASET` is initialized by `: ${ARROW_DATASET:=ON}`.)